lib/repositories/survey_repository.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:survey/models/survey_info.dart';
+import 'package:survey/services/api/survey/survey_api_service.dart';
+import 'package:survey/services/local_storage/local_storage_service.dart';
+import 'package:survey/services/locator/locator_service.dart';
+
+abstract class SurveyRepository {
+  static const listLocalStorageKey = "survey_repository_list";
+
+  Future<bool> get isSurveysCached;
+
+  Future<List<SurveyInfo>> fetchSurveysFromCached();
+
+  Future<List<SurveyInfo>> fetchSurveysFromRemote();
+}
+
+class SurveyRepositoryImpl implements SurveyRepository {
+  final SurveyApiService _surveyApiService = locator.get();
+  final LocalStorageService _localStorageService = locator.get();
+
+  @override
+  Future<bool> get isSurveysCached async {
+    final items = await _localStorageService
+        .getListObject<SurveyInfo>(SurveyRepository.listLocalStorageKey);
+    return items != null;
+  }
+
+  @override
+  Future<List<SurveyInfo>> fetchSurveysFromCached() async {
+    final items = await _localStorageService
+        .getListObject<SurveyInfo>(SurveyRepository.listLocalStorageKey);
+
+    return items ?? [];
+  }
+
+  @override
+  Future<List<SurveyInfo>> fetchSurveysFromRemote() async {
+    // Fetch from server
+    final list = await _surveyApiService.list(params: SurveyListParams());
+
+    // Storage it
+    _localStorageService.setListObject(
+      list.items,
+      key: SurveyRepository.listLocalStorageKey,
+    );
```

Comment:
hmm, I would not recommend to save large json data to SharedPreferences, SharedPreferences is not a place to store big objects but small and key values. Imagine you receive a huge json data, SharedPreferences won't allow you to save it. I think we should save surveys to database instead 🤔

Reply:
@manh-t  exactly, I just want to keep it simple lol
But if it's important I can change it later, at my final sprint 😁

Reply:
@markgravity I'm looking forward to see it 🥇